src/main/java/com/aws/greengrass/deployment/activator/DefaultActivator.java

```diff
@@ -59,7 +64,20 @@ public void activate(Map<String, Object> newConfig, Deployment deployment,
         // Get the timestamp before updateMap(). It will be used to check whether services have started.
         long mergeTime = System.currentTimeMillis();
 
-        updateConfiguration(deploymentDocument.getTimestamp(), newConfig);
+        // default to using the deployment creation time as the timestamp
+        long timestamp = deploymentDocument.getTimestamp();
+        Object deploymentConfigurationTimeSourceObject = kernel.getConfig().lookup(
+                SERVICES_NAMESPACE_TOPIC,
+                DeviceConfiguration.DEFAULT_NUCLEUS_COMPONENT_NAME,
+                CONFIGURATION_CONFIG_KEY,
+                DEVICE_PARAM_DEPLOYMENT_CONFIGURATION_TIME_SOURCE).getOnce();
+        if (deploymentConfigurationTimeSourceObject != null
+                && Coerce.toString(deploymentConfigurationTimeSourceObject)
+                .equals(DEPLOYMENT_CONFIGURATION_TIME_SOURCE_DEPLOYMENT_PROCESSING_TIME)) {
```

Comment:
Nit: Flip this comparison.

Reply:
Addressed via https://github.com/aws-greengrass/aws-greengrass-nucleus/pull/1706#discussion_r2052990079